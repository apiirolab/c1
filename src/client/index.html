<!DOCTYPE html>
<html>

<head>
  <title></title>
  <meta name="description" content="Corporate process management for Western Sydney University!">
  <link rel="icon" type="image/png" href="/favico.png">
  <link href='https://cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.3.2/semantic.min.css' rel='stylesheet' type='text/css'>
</head>

<body>
  <div id="root"></div>
  $bundles
</body>

</html>