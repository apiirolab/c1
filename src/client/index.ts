export const info = 'Client Info';
