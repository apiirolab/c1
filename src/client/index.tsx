import * as React from 'react';
import * as ReactDOM from 'react-dom';

import { ApolloProvider } from 'react-apollo';
import { Routes } from './config/router';
import { client } from './config/apollo';
// import * as serviceWorker from './registerServiceWorker';

ReactDOM.render(
  <ApolloProvider client={client}>
    <Routes />
  </ApolloProvider>,
  document.getElementById('root')
);

// If you want your app to work offline and load faster, you can change
// unregister() to register() below. Note this comes with some pitfalls.
// Learn more about service workers: http://bit.ly/CRA-PWA
// serviceWorker.unregister();
