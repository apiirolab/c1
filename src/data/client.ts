import * as QueryTypes from './generated/types';

export { default as gql } from 'graphql-tag';

export { QueryTypes };

export function purge<T>(params: any): T {
  for (let key of Object.getOwnPropertyNames(params)) {
    if (!params[key]) {
      delete params[key];
    }
    if (params[key] instanceof Object) {
      purge(params[key]);
    }
  }
  return params;
}
