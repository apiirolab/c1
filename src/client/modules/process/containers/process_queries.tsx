import * as PROCESSES_QUERY from 'data/bpmn_process/client/processes_query.graphql';

import { Query, QueryProps } from 'react-apollo';

import { Yoga } from 'data/yoga';

export { PROCESSES_QUERY };

/* =========================================================
    Notifications
   ======================================================== */

type Data = { processes: Yoga.BpmnProcess[] };
type Variables = { input: Yoga.BpmnProcessesInput };
type Props = Partial<QueryProps<Data, Variables>>;

export class ProcessListQuery extends Query<Data, Variables> {
  static displayName = 'ProcessesQuery';
  static defaultProps: Props = {
    query: PROCESSES_QUERY,
    variables: {
      input: {
        first: 50
      }
    }
  };
}
