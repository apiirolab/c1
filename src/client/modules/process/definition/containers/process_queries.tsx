import * as PROCESS_LIST_QUERY from '../queries/bpmn_processes_query.graphql';

import { Query, QueryProps } from 'react-apollo';

import { QueryTypes } from 'data/client';

export { PROCESS_LIST_QUERY };

/* =========================================================
    Process List
   ======================================================== */

type Data = QueryTypes.ProcessesQuery;
type Variables = QueryTypes.ProcessesQueryVariables;
type Props = Partial<QueryProps<QueryTypes.ProcessesQuery, QueryTypes.ProcessesQueryVariables>>;

export class ProcessListQuery extends Query<Data, Variables> {
  static displayName = 'ProcessesQuery';
  static defaultProps: Props = {
    query: PROCESS_LIST_QUERY,
    variables: {
      input: {
        first: 50
      }
    }
  };
}
