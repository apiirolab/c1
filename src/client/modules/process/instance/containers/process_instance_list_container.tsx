import * as React from 'react';

import { inject } from 'mobx-react';

import { renderResult } from 'client/modules/common';
import { purge, QueryTypes } from 'data/client';
import { ProcessInstanceList } from '../views/process_instance_list_view';
import { ProcessInstanceListQuery } from './process_instance_queries';

type Props = {
  path?: string;
  context?: App.Context;
  header: string;
  icon: string;
  description: string;
  status?: QueryTypes.BpmnProcessInstanceStatus;
};

export const ProcessInstanceListContainer: React.SFC<Props> = inject('context')(
  ({ context, header, icon, description, status }) => (
    <ProcessInstanceListQuery
      variables={purge<QueryTypes.BpmnProcessInstancesQueryVariables>({ input: { status } })}
    >
      {result => {
        // console.log(result.data.bpmnProcessInstances)
        return renderResult(result, () => (
          <ProcessInstanceList
            processes={result.data && result.data.bpmnProcessInstancesQuery}
            header={header}
            icon={icon}
            description={description}
            context={context}
          />
        ));
      }}
    </ProcessInstanceListQuery>
  )
);

ProcessInstanceListContainer.displayName = 'ProcessListContainer';
