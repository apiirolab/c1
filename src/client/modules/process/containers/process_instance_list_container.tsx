import * as React from 'react';

import { inject } from 'mobx-react';

import { renderResult } from 'client/modules/common';
import { ProcessInstanceList } from '../views/process_instance_list_view';
import { ProcessInstanceListQuery } from './process_queries';

type Props = {
  path?: string;
  context?: App.Context;
  header: string;
  icon: string;
  description: string;
};

export const ProcessListContainer: React.SFC<Props> = inject('context')(
  ({ context, header, icon, description }) => (
    <ProcessInstanceListQuery>
      {result => {
        // console.log(result.data.bpmnProcessInstances)
        return renderResult(result, () => (
          <ProcessInstanceList
            processes={result.data.bpmnProcessInstances}
            header={header}
            icon={icon}
            description={description}
            context={context}
          />
        ));
      }}
    </ProcessInstanceListQuery>
  )
);

ProcessListContainer.displayName = 'ProcessListContainer';
