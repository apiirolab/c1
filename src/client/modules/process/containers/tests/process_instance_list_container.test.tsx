import * as React from 'react';

import { mock, MockedProvider, render } from 'client/tests';

import { ProcessInstanceListContainer } from '../process_instance_list_container';
import { PROCESS_INSTANCE_LIST_QUERY } from '../process_queries';
import { createProcessInstanceList } from './process_test_data';

describe('Process Instance', () => {
  describe('List', () => {
    describe('Container', () => {
      beforeEach(() => mock.reset());

      function componentWithData() {
        return (
          <MockedProvider>
            <div>
              <ProcessInstanceListContainer
                icon="tasks"
                header="Process List"
                description="Process list description"
              />
            </div>
          </MockedProvider>
        );
      }

      function luisComponent() {
        mock.reset();
        mock.expect(PROCESS_INSTANCE_LIST_QUERY).reply({
          bpmnProcessInstancesQuery: createProcessInstanceList()
        });

        return componentWithData();
      }

      it('renders loading', () => {
        mock.expect(PROCESS_INSTANCE_LIST_QUERY).loading();
        const root = render(componentWithData());
        expect(root).toMatchSnapshot();
      });

      it('renders error', () => {
        mock.expect(PROCESS_INSTANCE_LIST_QUERY).fail('This is some error');
        const root = render(componentWithData());
        expect(root).toMatchSnapshot();
      });

      it('renders data', async () => {
        mock.expect(PROCESS_INSTANCE_LIST_QUERY).reply({
          bpmnProcessInstancesQuery: createProcessInstanceList()
        });

        const root = render(componentWithData());
        expect(root).toMatchSnapshot();
      });

      return {
        component: luisComponent()
      };
    });
  });
});
