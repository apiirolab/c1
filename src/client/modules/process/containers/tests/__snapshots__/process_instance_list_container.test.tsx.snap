// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Process Instances Container renders data 1`] = `
<div>
  <div
    className="sc-ifAKCX fUQyyW"
  >
    5 processes available
  </div>
  <div
    className="ui header"
  >
    <i
      aria-hidden="true"
      className="tasks icon"
    />
    <div
      className="content"
    >
      Process List
      <div
        className="sub header"
      >
        Process list description
      </div>
    </div>
  </div>
  <div
    className="ui divided relaxed list sc-bxivhb fuOSAw"
    role="list"
  >
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      <i
        aria-hidden="true"
        className="money icon middle aligned sc-bdVaJa jLAMJN"
      />
      <div
        className="top aligned content"
      >
        <a
          className="header"
          href="/nowhere"
          onClick={[Function]}
        >
          Process 1
        </a>
        <div
          className="description"
        >
          <div
            className="sc-bwzfXH gzggMh"
          >
            <div
              className="ui green tiny label"
              onClick={[Function]}
            >
              Running
            </div>
             
             · 
            Started 2 months ago by Buddy Holly
          </div>
          <div
            className="sc-htpNat eaijOI"
          >
            <b>
              Last activity: 
            </b>
            Tomas Trescak performed "Do not display!" 2 months ago
          </div>
          <div
            className="sc-htpNat eaijOI"
          >
            <b>
              Waiting for: 
            </b>
             
            Purchasing to Complete Purchase since 2 months ago and IT Support or IT Delivery to Deliver Technology since 2 months ago
          </div>
        </div>
      </div>
    </div>
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      <i
        aria-hidden="true"
        className="money icon middle aligned sc-bdVaJa jLAMJN"
      />
      <div
        className="top aligned content"
      >
        <a
          className="header"
          href="/nowhere"
          onClick={[Function]}
        >
          Process 1
        </a>
        <div
          className="description"
        >
          <div
            className="sc-bwzfXH gzggMh"
          >
            <div
              className="ui green tiny label"
              onClick={[Function]}
            >
              Running
            </div>
             
             · 
            Started 2 months ago by Buddy Holly
          </div>
          <div
            className="sc-htpNat eaijOI"
          >
            <b>
              Waiting for: 
            </b>
             
            Approver to First Task since 2 months ago
          </div>
        </div>
      </div>
    </div>
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      <i
        aria-hidden="true"
        className="money icon middle aligned sc-bdVaJa jLAMJN"
      />
      <div
        className="top aligned content"
      >
        <a
          className="header"
          href="/nowhere"
          onClick={[Function]}
        >
          Process 1
        </a>
        <div
          className="description"
        >
          <div
            className="sc-bwzfXH gzggMh"
          >
            <div
              className="ui red tiny label"
              onClick={[Function]}
            >
              Aborted
            </div>
             
             · 
            Started 2 months ago by Buddy Holly
          </div>
          <div
            className="sc-htpNat eaijOI"
          >
            <b>
              Last activity: 
            </b>
            Tomas Trescak performed "Do not display!" 2 months ago
          </div>
        </div>
      </div>
    </div>
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      <i
        aria-hidden="true"
        className="money icon middle aligned sc-bdVaJa jLAMJN"
      />
      <div
        className="top aligned content"
      >
        <a
          className="header"
          href="/nowhere"
          onClick={[Function]}
        >
          Process 1
        </a>
        <div
          className="description"
        >
          <div
            className="sc-bwzfXH gzggMh"
          >
            <div
              className="ui yellow tiny label"
              onClick={[Function]}
            >
              Paused
            </div>
             
             · 
            Started 2 months ago by Buddy Holly
          </div>
          <div
            className="sc-htpNat eaijOI"
          >
            <b>
              Last activity: 
            </b>
            Tomas Trescak performed "Do not display!" 2 months ago
          </div>
        </div>
      </div>
    </div>
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      <i
        aria-hidden="true"
        className="money icon middle aligned sc-bdVaJa jLAMJN"
      />
      <div
        className="top aligned content"
      >
        <a
          className="header"
          href="/nowhere"
          onClick={[Function]}
        >
          Process 1
        </a>
        <div
          className="description"
        >
          <div
            className="sc-bwzfXH gzggMh"
          >
            <div
              className="ui grey tiny label"
              onClick={[Function]}
            >
              Finished
            </div>
             
             · 
            Started 2 months ago by Buddy Holly
             · 
            Finished 2 months ago
          </div>
          <div
            className="sc-htpNat eaijOI"
          >
            <b>
              Last activity: 
            </b>
            Tomas Trescak performed "Last Task" 2 months ago
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Process Instances Container renders error 1`] = `
<div>
  <div>
    <div
      className="ui error message"
    >
      <div
        className="content"
      >
        <div
          className="header"
        >
          👾 Unexpected Error. We apologise for any inconvenience. Please contact your admins with the message below.
        </div>
        <div
          className="ui list"
          role="list"
        >
          <div
            className="item"
            onClick={[Function]}
            role="listitem"
          >
            This is some error
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Process Instances Container renders loading 1`] = `
<div>
  <div>
    <div
      className="ui tiny active inline loader"
    />
     
     Loading ...
  </div>
</div>
`;

exports[`Process Instances List Container renders data 1`] = `
<div>
  <div
    className="sc-ifAKCX fUQyyW"
  >
    5 processes available
  </div>
  <div
    className="ui header"
  >
    <i
      aria-hidden="true"
      className="tasks icon"
    />
    <div
      className="content"
    >
      Process List
      <div
        className="sub header"
      >
        Process list description
      </div>
    </div>
  </div>
  <div
    className="ui divided relaxed list sc-bxivhb fuOSAw"
    role="list"
  >
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      <i
        aria-hidden="true"
        className="money icon middle aligned sc-bdVaJa jLAMJN"
      />
      <div
        className="top aligned content"
      >
        <a
          className="header"
          href="/nowhere"
          onClick={[Function]}
        >
          Process 1
        </a>
        <div
          className="description"
        >
          <div
            className="sc-bwzfXH gzggMh"
          >
            <div
              className="ui green tiny label"
              onClick={[Function]}
            >
              Running
            </div>
             
             · 
            Started 2 months ago by Buddy Holly
          </div>
          <div
            className="sc-htpNat eaijOI"
          >
            <b>
              Last activity: 
            </b>
            Tomas Trescak performed "Do not display!" 2 months ago
          </div>
          <div
            className="sc-htpNat eaijOI"
          >
            <b>
              Waiting for: 
            </b>
             
            Purchasing to Complete Purchase since 2 months ago and IT Support or IT Delivery to Deliver Technology since 2 months ago
          </div>
        </div>
      </div>
    </div>
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      <i
        aria-hidden="true"
        className="money icon middle aligned sc-bdVaJa jLAMJN"
      />
      <div
        className="top aligned content"
      >
        <a
          className="header"
          href="/nowhere"
          onClick={[Function]}
        >
          Process 1
        </a>
        <div
          className="description"
        >
          <div
            className="sc-bwzfXH gzggMh"
          >
            <div
              className="ui green tiny label"
              onClick={[Function]}
            >
              Running
            </div>
             
             · 
            Started 2 months ago by Buddy Holly
          </div>
          <div
            className="sc-htpNat eaijOI"
          >
            <b>
              Waiting for: 
            </b>
             
            Approver to First Task since 2 months ago
          </div>
        </div>
      </div>
    </div>
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      <i
        aria-hidden="true"
        className="money icon middle aligned sc-bdVaJa jLAMJN"
      />
      <div
        className="top aligned content"
      >
        <a
          className="header"
          href="/nowhere"
          onClick={[Function]}
        >
          Process 1
        </a>
        <div
          className="description"
        >
          <div
            className="sc-bwzfXH gzggMh"
          >
            <div
              className="ui red tiny label"
              onClick={[Function]}
            >
              Aborted
            </div>
             
             · 
            Started 2 months ago by Buddy Holly
          </div>
          <div
            className="sc-htpNat eaijOI"
          >
            <b>
              Last activity: 
            </b>
            Tomas Trescak performed "Do not display!" 2 months ago
          </div>
        </div>
      </div>
    </div>
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      <i
        aria-hidden="true"
        className="money icon middle aligned sc-bdVaJa jLAMJN"
      />
      <div
        className="top aligned content"
      >
        <a
          className="header"
          href="/nowhere"
          onClick={[Function]}
        >
          Process 1
        </a>
        <div
          className="description"
        >
          <div
            className="sc-bwzfXH gzggMh"
          >
            <div
              className="ui yellow tiny label"
              onClick={[Function]}
            >
              Paused
            </div>
             
             · 
            Started 2 months ago by Buddy Holly
          </div>
          <div
            className="sc-htpNat eaijOI"
          >
            <b>
              Last activity: 
            </b>
            Tomas Trescak performed "Do not display!" 2 months ago
          </div>
        </div>
      </div>
    </div>
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      <i
        aria-hidden="true"
        className="money icon middle aligned sc-bdVaJa jLAMJN"
      />
      <div
        className="top aligned content"
      >
        <a
          className="header"
          href="/nowhere"
          onClick={[Function]}
        >
          Process 1
        </a>
        <div
          className="description"
        >
          <div
            className="sc-bwzfXH gzggMh"
          >
            <div
              className="ui grey tiny label"
              onClick={[Function]}
            >
              Finished
            </div>
             
             · 
            Started 2 months ago by Buddy Holly
             · 
            Finished 2 months ago
          </div>
          <div
            className="sc-htpNat eaijOI"
          >
            <b>
              Last activity: 
            </b>
            Tomas Trescak performed "Last Task" 2 months ago
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Process Instances List Container renders error 1`] = `
<div>
  <div>
    <div
      className="ui error message"
    >
      <div
        className="content"
      >
        <div
          className="header"
        >
          👾 Unexpected Error. We apologise for any inconvenience. Please contact your admins with the message below.
        </div>
        <div
          className="ui list"
          role="list"
        >
          <div
            className="item"
            onClick={[Function]}
            role="listitem"
          >
            This is some error
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Process Instances List Container renders loading 1`] = `
<div>
  <div>
    <div
      className="ui tiny active inline loader"
    />
     
     Loading ...
  </div>
</div>
`;
